src/app/core/upload/upload.service.ts

```diff
@@ -86,41 +96,40 @@ export class Upload {
     this.upload();
   }
 
-  cancel(): boolean {
-    if (this.evaporate && !this.complete) {
-      let formerId = this.uploadId;
-      this.uploadId = null;
-      if (formerId !== null) {
-        return this.evaporate.cancel(formerId);
-      }
+  cancel(): Observable<any> {
+    if (this.evaporate && !this.complete && this.uploadId) {
+      return this.uploadId.map(id => {
+        this.uploadId = null;
+        return this.evaporate.cancel(Env.BUCKET_NAME + '/' + id);
+      });
     }
-    return false;
+    return Observable.of(false);
   }
 
   upload(): Observable<number> {
     if (this.complete) {
-      return null;
+      return Observable.of(null);
     }
     this.cancel();
 
-    let uploadOptions = {
+    const uploadOptions = {
       file: this.file,
       name: this.path,
       contentType: this.contentType,
       xAmzHeadersAtInitiate: {
         'x-amz-acl': 'public-read'
       },
       notSignedHeadersAtInitiate: {
-        'Content-Disposition': 'attachment; filename=' + name
+        'Content-Disposition': 'attachment; filename=' + this.name
       }
     };
 
-    let progressObservable: Observable<number> = Observable.create((sub: Subscriber<number>) => {
+    const progressObservable: Observable<number> = Observable.create((sub: Subscriber<number>) => {
       sub.next(0);
       uploadOptions['progress'] = (pct: number) => sub.next(pct);
       uploadOptions['complete'] = () => { sub.next(1.0); this.complete = true; sub.complete(); };
       uploadOptions['error']    = (msg: string) => sub.error(msg);
-      this.uploadId = this.evaporate.add(uploadOptions);
+      this.uploadId = Observable.from(this.evaporate.add(uploadOptions));
```

Comment:
I'm not sure this is working as expected.  When I click "cancel", I'm not seeing `this.evaporate.cancel` ever get called, maybe because this Observable isn't emitting anything.

Reply:
Nvm ... it does eventually get there.  But only after the upload completes.  May need some other mechanism for cancelling then?